File: Examples/VisualBasic/AsposePDF/Forms/FillXFAFields.vb
Imports System.IO
Imports System
Imports Microsoft.VisualBasic
Imports Aspose.Pdf

Namespace AsposePDF.Forms
    Public Class FillXFAFields
        Public Shared Sub Run()
            ' ExStart:FillXFAFields
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_AsposePdf_Forms()

            ' Load XFA form
            Dim doc As New Document(dataDir & Convert.ToString("FillXFAFields.pdf"))

            ' Get names of XFA form fields
            Dim names As String() = doc.Form.XFA.FieldNames

            ' Set field values
            doc.Form.XFA(names(0)) = "Field 0"
            doc.Form.XFA(names(1)) = "Field 1"
            dataDir = dataDir & Convert.ToString("Filled_XFA_out_.pdf")
            ' Save the updated document
            doc.Save(dataDir)
            ' ExEnd:FillXFAFields
            Console.WriteLine(Convert.ToString(vbLf & "XFA fields filled successfully." & vbLf & "File saved at ") & dataDir)
        End Sub
    End Class
End Namespace

File: Examples/CSharp/AsposePDF/Forms/FillXFAFields.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2013 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;

namespace CSharp.AsposePdf.Forms
{
    public class FillXFAFields
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdf_Forms();

            // load XFA form
            Document doc = new Document(dataDir + "FillXFAFields.pdf");

            // get names of XFA form fields
            string[] names = doc.Form.XFA.FieldNames;

            //set field values
            doc.Form.XFA[names[0]] = "Field 0";
            doc.Form.XFA[names[1]] = "Field 1";

            // save the updated document
            doc.Save(dataDir + "Filled_XFA.pdf");

        }
    }
}